gammapy/estimators/flux_map.py

```diff
@@ -552,6 +574,18 @@ def eflux_ul(self):
         """Return energy flux (eflux) SED upper limits."""
         return self.norm_ul * self.eflux_ref
 
+    def _filter_convergence_failure(self, some_map):
+        """Put NaN where pixels did not converge."""
+        if not self.remove_failed:
+            return some_map
+
+        if (not self.has_success) or (not self.success.data.shape == some_map.data.shape):
```

Comment:
The implicit assumption is that the geometries of the maps are compatible in `FluxMaps`, so you could skip the second check here...

Reply:
In the function, flux_point::from_table, there is a look on each individual map. The same occurs for the run function of the lightcurve estimator. During this loop, one map is reduced but not yet the success map. Without this second check, the lign `new_map.data[~self.success.data] = np.nan` will lead to a crash.

Reply:
My first thought here is, that the `success` map, **must** have the same dimension and axes as the `norm` map, because each norm pixel is associated to a fit. This is independent of `FluxPoints` or `FluxMaps`. I still haven't understood, why the second would be needed.

Reply:
OK, now I think I understand: the failures you see are related to the `counts`, `npred` and `stat_scan` quantities. Those are the quantities, that have a different dimension than `norm`. They either include an additional `norm` or `dataset` axis. In this case the approach of:
```python3
new_map.data[~self.success.data] = np.nan
```
Does not work and raises an error, because the dimension do not agree. The solution here is to rely on Numpy's broadcasting instead, e.g. via multiplication:
```python3
mask_nan = np.ones(success.data.shape)
mask_nan[~success.data] = np.nan
new_map = some_map * mask_nan
```

Reply:
On the other hand it might not make sense at all to filter the `counts`, `npred` and `stat_scan` quantities, because those are the quantities one would like to check, when the fit failed...but I guess that's why we introduced the `filter_success_nan` flag.